### PR TITLE
docs(headless): improve automatic facet doc to reflect what should be used

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/KIT-2838

The docs are not great and even my solution is not the best. The best would be to actually remove the documentation around the `buildAutomaticFacet` and the `AutomaticFacetProps`. The reason why is because they are not supposed to be used by clients, only internally by the generator. Unfortunately, we cant remove them from the docs because they are needed for the rest of the docs to generate... 

To properly solve this, we would need to refactor the doc-parser to enable _ignoring_ certain function/interfaces.

### CODE REVIEW

File: packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.test.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+} from '../../../controllers/facets/automatic-facet/headless-automatic-facet';
 import {
   deselectAllAutomaticFacetValues,
   toggleSelectAutomaticFacetValue,
 } from '../../../features/facets/automatic-facet-set/automatic-facet-set-actions';
 import {executeSearch} from '../../../features/search/search-actions';
+import {SearchAppState} from '../../../state/search-app-state';
 import {
   MockSearchEngine,
   buildMockSearchAppEngine,
```
Comment (alexprudhomme) [inline packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.test.ts:13]: Just something random I came across where we were importing from the index.ts

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts [hunk 2]
```diff
@@ -40,14 +40,14 @@ export interface AutomaticFacetProps {
 }
 /**
  * The `AutomaticFacet` controller allows you to create a search interface component that the end user
- * can use to refine a query by selecting filters based on item metadata (i.e., field values). Unlike traditional facets that
+ * can use to refine a query by selecting filters based on item metadata (i.e., field values). Unlike regular facets that
  * need to be explicitly defined and requested in the query, automatic facets are dynamically generated by the index
  * in response to the query.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts:45]: The `AutomaticFacet` controller is used internally by the `AutomaticFacetGenerator` controller to manage the individual state of each dynamically generated facet.

To learn more about the automatic facet generator feature, see [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).

**Important:** This controller is meant for internal use by Headless only. As an implementer, you should never import or use this controller directly in your code.

File: packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts [hunk 3]
```diff
@@ -66,10 +66,10 @@ export interface AutomaticFacet extends Controller {
   state: AutomaticFacetState;
 }
 /**
- * Creates a `AutomaticFacet` controller instance.
+ * This initializer is used internally by the AutomaticFacetGenerator controller.
  *
  * @param engine - The headless engine.
- * @param props - The configurable `AutomaticFacet` properties.
+ * @param props - The configurable `AutomaticFacet` properties used internally.
  * @returns A `AutomaticFacet` controller instance.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts:73]:
```suggestion
 * @returns An `AutomaticFacet` controller instance.
```

File: packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts [hunk 4]
```diff
@@ -66,10 +66,10 @@ export interface AutomaticFacet extends Controller {
   state: AutomaticFacetState;
 }
 /**
- * Creates a `AutomaticFacet` controller instance.
+ * This initializer is used internally by the AutomaticFacetGenerator controller.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts:69]:
```suggestion
 * This initializer is used internally by the `AutomaticFacetGenerator` controller.
 *
 * **Important:** This initializer is meant for internal use by Headless only. As an implementer, you should never import or use this initializer directly in your code.
```

File: packages/samples/headless-react/src/components/automatic-facet/automatic-facet.fn.tsx [hunk 5]
```diff
@@ -28,3 +28,9 @@ export const AutomaticFacet: FunctionComponent<AutomaticFacetProps> = (
     </ul>
   );
 };
+
```
Comment (fbeaudoincoveo) [inline packages/samples/headless-react/src/components/automatic-facet/automatic-facet.fn.tsx:31]: Might as well just remove the sample if this is meant for internal use only.

**Review by fbeaudoincoveo — CHANGES_REQUESTED** [review]
I left some suggestions to rephrase the documentation, but why do you even need to export AtomaticFacet, AutomaticFacetProps, and AutomaticFacetState in index.ts if this controller is 100% internal?

If you don't export it, you could just remove the docs.

If the controller was ALWAYS intended to be internal and should realisitically have been used in no implementations, then removing the export would not be a breaking change IMO.

**Comment by alexprudhomme** [comment]
@fbeaudoincoveo I made some changes, we now only export the AutomaticFacet and AutomaticFacetState interfaces that are important for implementers. They are now exported inside of the Generator page directly. There wont be a doc page for AutomaticFacet, only one for the Generator !

**Review by srosenberg-coveo — APPROVED** [review]

File: packages/headless/src/controllers/facets/automatic-facet-generator/headless-automatic-facet-generator.ts [hunk 6]
```diff
@@ -53,6 +52,48 @@ export interface AutomaticFacetGeneratorProps {
...
+   */
+  label: string;
+  /**
+   * The automatic facet values.
+   */
+  values: FacetValue[];
+}
+/**
+ * The `AutomaticFacet` controller allows you to create a search interface component that the end user
+ * can use to refine a query by selecting filters based on item metadata (i.e., field values). Unlike regular facets that
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/facets/automatic-facet-generator/headless-automatic-facet-generator.ts:74]:
```suggestion
 * can interact with to refine a query by selecting filters based on item metadata (i.e., field values). Unlike regular facets that
```

pet peeve, but I just don't like "that the end _user_ can _use_" 😅

File: packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts [hunk 7]
```diff
@@ -7,70 +7,29 @@ import {logFacetClearAll} from '../../../features/facets/facet-set/facet-set-ana
...
-   */
-  label: string;
-  /**
-   * The automatic facet values.
-   */
-  values: FacetValue[];
-}
-/**
- * The props for the `AutomaticFacet` controller.
+ * @internal
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts:14]: If the props / builder aren't exported, do we even need to document them?

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts (hunk 7 above)
Comment (alexprudhomme) [reply packages/headless/src/controllers/facets/automatic-facet/headless-automatic-facet.ts:14]: I think its fine to leave the comments.

**Pull request merged**